source/strings.tex

```diff
@@ -6402,3 +6488,89 @@
+it determines the values of the corresponding UTF-8 code units and then,
+if \tcode{pc8} is not a null pointer,
+stores the value of the first (or only) such code unit
+in the object pointed to by \tcode{pc8}.
+Subsequent calls will store successive UTF-8 code units
+without consuming any additional input
+until all the code units have been stored.
+If the corresponding Unicode character is \tcode{U+0000},
+the resulting state described is the initial conversion state.
+
+\pnum
+\returns
+The first of the following that applies (given the current conversion state):
+\begin{itemize}
+\item \tcode{0}, if the next \tcode{n} or fewer bytes complete
+the multibyte character
+that corresponds to the \tcode{U+0000} Unicode character
+(which is the value stored).
+\item between \tcode{1} and \tcode{n} inclusive,
+if the next n or fewer bytes complete a valid multibyte character
+(which is the value stored);
+the value returned is the number of bytes that complete the multibyte character.
+\item \tcode{(size_t)(-3)}, if the next character
+resulting from a previous call has been stored
+(no bytes from the input have been consumed by this call).
+\item \tcode{(size_t)(-2)}, if the next \tcode{n} bytes
+contribute to an incomplete (but potentially valid) multibyte character, and
+all \tcode{n} bytes have been processed (no value is stored).
+\item \tcode{(size_t)(-1)}, if an encoding error occurs,
+in which case the next \tcode{n} or fewer bytes do not contribute to
+a complete and valid multibyte character (no value is stored);
+the value of the macro \tcode{EILSEQ} is stored in \tcode{errno}, and
+the conversion state is unspecified.
+\end{itemize}
+\end{itemdescr}
+
+\begin{itemdecl}
+size_t c8rtomb(char* s, char8_t c8, mbstate_t* ps);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+If \tcode{s} is a null pointer, equivalent to
+\tcode{c8rtomb(buf, u8'$\backslash$0', ps)}
+where \tcode{buf} is an internal buffer.
+Otherwise, if \tcode{c8} completes a sequence of valid UTF-8 code units,
+determines the number of bytes needed
+to represent the multibyte character (including any shift sequences),
+and stores the multibyte character representation in the array
+whose first element is pointed to by \tcode{s}.
+At most \tcode{MB_CUR_MAX} bytes are stored.
+If the multibyte character is a null character, a null byte is stored,
+preceded by any shift sequence needed to restore the initial shift state;
+the resulting state described is the initial conversion state.
+
+\pnum
+\returns
+The number of bytes stored in the array object (including any shift sequences).
+If \tcode{c8} does not contribute to a sequence of \tcode{char8_t}
+corresponding to a valid multibyte character,
+the value of the macro \tcode{EILSEQ} is stored in \tcode{errno},
+\tcode{(size_t) (−1)} is returned, and the conversion state is unspecified.
+
+\pnum
+\remarks
+Calls to \tcode{c8rtomb} with a null pointer argument for \tcode{buf}
```

Comment:
There is no parameter named "buf". I think this "buf" should be "s" (same 2 lines lower).

Reply:
Right, fixed.